### PR TITLE
feat(utility): support engine.online to start online compute mode

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-32bcurb 支持 engine.online() 来开启在线计算](https://app.clickup.com/t/32bcurb)

File: graviti/utility/engine.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class Online:
+    """An engine controller used to start and stop the online mode."""
+
+    enabled: bool = False
+
+    def __init__(self, engine_: "Engine") -> None:
+        self.engine = engine_
+
+    def __enter__(self) -> None:
+        self.engine.mode = "online"
```
Comment (linjiX) [inline graviti/utility/engine.py:20]: do we need a enum for the online and local?

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -774,6 +774,9 @@ def query(self, func: Callable[[Any], Any]) -> "DataFrame":
             0   a.jpg    1      1
 
         """
+        if engine.mode != "online":
+            raise TypeError("Only online mode is supported now.")
```
Comment (linjiX) [inline graviti/dataframe/frame.py:778]: do we need to mention how to use online mode in the error message?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/engine.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+"""Engine control related classes."""
+
+from typing import Any
+
+
+class Online:
+    """An engine controller used to start and stop the online mode."""
+
+    enabled: bool = False
```
Comment (linjiX) [inline graviti/utility/engine.py:14]: why we nee this?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/utility/engine.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+class Online:
+    """An engine controller used to start and stop the online mode."""
+
+    pre_mode: Mode
+
+    def __init__(self, engine_: "Engine") -> None:
+        self.engine = engine_
+
+    def __enter__(self) -> None:
+        self.pre_mode = self.engine.mode
```
Comment (linjiX) [inline graviti/utility/engine.py:28]: Maybe `old_mode` is better?
also this variable should be private.

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**